fastlane/fastlane
# [spaceship] allow 2FA request only in interactive mode

### Checklist

### Motivation and Context
Under company wide policy it's impossible to create an account without 2FA on Apple developer portal. At the same time, we use `spaceauth` on our CI and once the session expires, the job hangs until it times out instead of failing immediately. As I don't see ways to automate process of 2FA on CI I rather prefer it to fail immediately in non-interactive mode.

### Description
Check for `interactive` on the very first step after receiving `409` and raise an exception to fail the execution.
Due to [the lint rule](https://github.com/fastlane/fastlane/blob/e89cdcfb869ad0640438cc5f6ed261bb50887641/fastlane/Fastfile#L36) to avoid `FastlaneCore::UI` I duplicated the logic to determine interactive mode. If there's no clear reason not to import `fastlane_core` I can simplify this by removing this lint

Tests expect the exception in case non-interactive launch and no exception in interactive.

### Testing Steps
- Run `bundle exec fastlane spaceauth` with an account with 2FA. Make sure interactive console requests 2FA code.
- Run `export CI=1 && bundle exec fastlane spaceauth` with an account with 2FA. Make sure action fails without waiting for interaction.
